// true-food-bite-frontend/src/features/auth/pages/Login.jsx
import { useState, useEffect } from 'react';
import { Link, useNavigate, useLocation } from 'react-router-dom';
import { motion } from 'framer-motion';
import { Mail, Lock, Eye, EyeOff, LogIn, AlertCircle, CheckCircle, Scan, ShieldCheck, Leaf, Zap } from 'lucide-react';
import { useAuth } from '../context/AuthContext';

const FEATURES = [
    { icon: Scan, color: '#84cc16', text: 'Scan 3M+ verified products instantly' },
    { icon: ShieldCheck, color: '#22c55e', text: 'Allergen & additive alerts (FSSAI compliant)' },
    { icon: Leaf, color: '#0ea5e9', text: 'NOVA & Nutri-Grade A–E classification' },
    { icon: Zap, color: '#f59e0b', text: 'Personalised scan history & health dashboard' },
];

const Field = ({ label, icon: Icon, type, name, value, onChange, error, placeholder, right }) => (
    <div style={{ marginBottom: '1.25rem' }}>
        <label style={{ display: 'block', marginBottom: '0.45rem', fontSize: '0.85rem', fontWeight: '700', color: '#94a3b8' }}>{label}</label>
        <div style={{ position: 'relative' }}>
            <Icon size={17} style={{ position: 'absolute', left: '1rem', top: '50%', transform: 'translateY(-50%)', color: error ? '#f87171' : '#475569', pointerEvents: 'none' }} />
            <input
                type={type} name={name} value={value} onChange={onChange} placeholder={placeholder}
                style={{
                    width: '100%', padding: right ? '0.9rem 3rem 0.9rem 2.85rem' : '0.9rem 1rem 0.9rem 2.85rem',
                    background: 'rgba(255,255,255,0.04)', border: `1px solid ${error ? '#f87171' : 'rgba(255,255,255,0.09)'}`,
                    borderRadius: '14px', color: '#f1f5f9', fontSize: '0.95rem', outline: 'none',
                    boxSizing: 'border-box', transition: 'border-color 0.2s',
                }}
                onFocus={e => e.target.style.borderColor = error ? '#f87171' : 'rgba(132,204,22,0.5)'}
                onBlur={e => e.target.style.borderColor = error ? '#f87171' : 'rgba(255,255,255,0.09)'}
            />
            {right}
        </div>
        {error && <p style={{ marginTop: '0.35rem', fontSize: '0.78rem', color: '#f87171' }}>{error}</p>}
    </div>
);

const Login = () => {
    const navigate = useNavigate();
    const location = useLocation();
    const { login } = useAuth();

    const [formData, setFormData] = useState({ email: '', password: '' });
    const [showPassword, setShowPassword] = useState(false);
    const [errors, setErrors] = useState({});
    const [success, setSuccess] = useState('');
    const [isLoading, setIsLoading] = useState(false);

    useEffect(() => {
        if (location.state?.message) {
            setSuccess(location.state.message);
            setTimeout(() => setSuccess(''), 5000);
            window.history.replaceState({}, document.title);
        }
    }, [location]);

    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(prev => ({ ...prev, [name]: value }));
        if (errors[name]) setErrors(prev => ({ ...prev, [name]: '' }));
    };

    const handleSubmit = async (e) => {
        e.preventDefault();
        const newErrors = {};
        if (!formData.email) newErrors.email = 'Email is required';
        else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(formData.email)) newErrors.email = 'Enter a valid email address';
        if (!formData.password) newErrors.password = 'Password is required';
        if (Object.keys(newErrors).length > 0) { setErrors(newErrors); return; }

        setIsLoading(true); setErrors({});
        try {
            const result = await login(formData.email, formData.password);
            if (result.success) navigate('/');
            else setErrors({ general: result.error || 'Login failed. Please check your credentials.' });
        } catch {
            setErrors({ general: 'An unexpected error occurred. Please try again.' });
        } finally { setIsLoading(false); }
    };

    return (
        <div style={{ minHeight: 'calc(100vh - 80px)', display: 'flex' }}>
            {/* Left panel — info */}
            <div style={{
                flex: 1, background: 'linear-gradient(135deg, rgba(124,58,237,0.15) 0%, rgba(132,204,22,0.08) 100%)',
                borderRight: '1px solid rgba(255,255,255,0.06)', padding: '4rem 3rem', flexDirection: 'column', justifyContent: 'center'
            }} className="auth-left-panel">
                <div>
                    <div style={{ marginBottom: '3rem' }}>
                        <span style={{ fontSize: '1.8rem', fontWeight: '900', background: 'linear-gradient(135deg,#84cc16,#0ea5e9)', WebkitBackgroundClip: 'text', WebkitTextFillColor: 'transparent' }}>True FoodBite</span>
                        <p style={{ color: '#475569', fontSize: '0.85rem', marginTop: '0.4rem' }}>India's food transparency platform</p>
                    </div>
                    <h2 style={{ fontSize: '2rem', fontWeight: '900', color: '#f1f5f9', lineHeight: 1.15, marginBottom: '1.75rem' }}>
                        Your personal food<br />intelligence dashboard
                    </h2>
                    <div style={{ display: 'flex', flexDirection: 'column', gap: '1rem' }}>
                        {FEATURES.map(f => (
                            <div key={f.text} style={{ display: 'flex', alignItems: 'center', gap: '0.85rem', padding: '0.85rem 1.1rem', borderRadius: '14px', background: 'rgba(255,255,255,0.03)', border: '1px solid rgba(255,255,255,0.06)' }}>
                                <div style={{ width: '36px', height: '36px', borderRadius: '10px', background: `${f.color}18`, display: 'flex', alignItems: 'center', justifyContent: 'center', flexShrink: 0 }}>
                                    <f.icon size={18} color={f.color} />
                                </div>
                                <span style={{ fontSize: '0.85rem', color: '#94a3b8' }}>{f.text}</span>
                            </div>
                        ))}
                    </div>
                </div>
            </div>

            {/* Right panel — form */}
            <div style={{ flex: '0 0 auto', width: '100%', maxWidth: '500px', margin: '0 auto', display: 'flex', alignItems: 'center', padding: '2rem 1.5rem' }}>
                <motion.div initial={{ opacity: 0, y: 20 }} animate={{ opacity: 1, y: 0 }} transition={{ duration: 0.4 }} style={{ width: '100%' }}>
                    {/* Icon + title */}
                    <div style={{ textAlign: 'center', marginBottom: '2rem' }}>
                        <div style={{ width: '64px', height: '64px', borderRadius: '20px', background: 'linear-gradient(135deg,#7c3aed,#ec4899)', display: 'flex', alignItems: 'center', justifyContent: 'center', margin: '0 auto 1.25rem', boxShadow: '0 8px 24px rgba(124,58,237,0.35)' }}>
                            <LogIn size={30} color="#fff" />
                        </div>
                        <h1 style={{ fontSize: '2rem', fontWeight: '900', background: 'linear-gradient(135deg,#7c3aed,#ec4899)', WebkitBackgroundClip: 'text', WebkitTextFillColor: 'transparent', marginBottom: '0.4rem' }}>Welcome Back</h1>
                        <p style={{ color: '#64748b', fontSize: '0.9rem' }}>Sign in to your True FoodBite account</p>
                    </div>

                    {/* Form card */}
                    <div style={{ background: 'rgba(255,255,255,0.03)', border: '1px solid rgba(255,255,255,0.08)', borderRadius: '24px', padding: '2.25rem', backdropFilter: 'blur(16px)', boxShadow: '0 20px 60px rgba(0,0,0,0.4)' }}>
                        {success && (
                            <div style={{ padding: '0.85rem 1rem', background: 'rgba(34,197,94,0.08)', border: '1px solid rgba(34,197,94,0.3)', borderRadius: '12px', marginBottom: '1.25rem', display: 'flex', alignItems: 'center', gap: '0.65rem' }}>
                                <CheckCircle size={16} color="#22c55e" />
                                <span style={{ color: '#22c55e', fontSize: '0.85rem' }}>{success}</span>
                            </div>
                        )}
                        {errors.general && (
                            <div style={{ padding: '0.85rem 1rem', background: 'rgba(239,68,68,0.08)', border: '1px solid rgba(239,68,68,0.3)', borderRadius: '12px', marginBottom: '1.25rem', display: 'flex', alignItems: 'center', gap: '0.65rem' }}>
                                <AlertCircle size={16} color="#ef4444" />
                                <span style={{ color: '#ef4444', fontSize: '0.85rem' }}>{errors.general}</span>
                            </div>
                        )}

                        <form onSubmit={handleSubmit}>
                            <Field label="Email Address" icon={Mail} type="email" name="email" value={formData.email} onChange={handleChange} error={errors.email} placeholder="your@email.com" />
                            <Field
                                label="Password" icon={Lock} type={showPassword ? 'text' : 'password'}
                                name="password" value={formData.password} onChange={handleChange}
                                error={errors.password} placeholder="Enter your password"
                                right={(
                                    <button type="button" onClick={() => setShowPassword(!showPassword)} style={{ position: 'absolute', right: '1rem', top: '50%', transform: 'translateY(-50%)', background: 'none', border: 'none', cursor: 'pointer', color: '#475569', display: 'flex' }}>
                                        {showPassword ? <EyeOff size={17} /> : <Eye size={17} />}
                                    </button>
                                )}
                            />

                            <div style={{ textAlign: 'right', marginBottom: '1.5rem', marginTop: '-0.5rem' }}>
                                <Link to="/forgot-password" style={{ color: '#7c3aed', fontSize: '0.82rem', fontWeight: '700', textDecoration: 'none' }}>Forgot Password?</Link>
                            </div>

                            <button type="submit" disabled={isLoading} style={{
                                width: '100%', padding: '0.95rem', borderRadius: '14px', border: 'none',
                                background: isLoading ? 'rgba(124,58,237,0.4)' : 'linear-gradient(135deg,#7c3aed,#ec4899)',
                                color: '#fff', fontWeight: '800', fontSize: '1rem',
                                cursor: isLoading ? 'not-allowed' : 'pointer',
                                boxShadow: isLoading ? 'none' : '0 4px 20px rgba(124,58,237,0.4)',
                                transition: 'all 0.2s ease',
                            }}>
                                {isLoading ? 'Signing in...' : 'Sign In to True FoodBite'}
                            </button>
                        </form>

                        <div style={{ marginTop: '1.5rem', paddingTop: '1.5rem', borderTop: '1px solid rgba(255,255,255,0.06)', textAlign: 'center' }}>
                            <p style={{ color: '#475569', fontSize: '0.88rem' }}>
                                Don't have an account?{' '}
                                <Link to="/signup" style={{ color: '#7c3aed', fontWeight: '700', textDecoration: 'none' }}>Create one free →</Link>
                            </p>
                        </div>
                    </div>
                    <p style={{ textAlign: 'center', marginTop: '1.25rem', fontSize: '0.72rem', color: '#334155' }}>
                        By signing in you agree to our <Link to="/terms-of-service" style={{ color: '#475569', textDecoration: 'underline' }}>Terms</Link> and <Link to="/privacy-policy" style={{ color: '#475569', textDecoration: 'underline' }}>Privacy Policy</Link>
                    </p>
                </motion.div>
            </div>

            <style>{`.auth-left-panel { display: none; } @media (min-width: 900px) { .auth-left-panel { display: flex !important; } }`}</style>
        </div>
    );
};

export default Login;
