// true-food-bite-frontend/src/features/core/components/Navbar.jsx
import { Link } from 'react-router-dom';
import { Scan, Sun, Moon, User, LogOut, Layout } from 'lucide-react';
import { useTheme } from './ThemeContext';
import { useAuth } from '../../auth/context/AuthContext';
import { useState, useEffect } from 'react';

// Fast live clock hook
function useLiveClock() {
    const [now, setNow] = useState(new Date());
    useEffect(() => {
        const id = setInterval(() => setNow(new Date()), 1000);
        return () => clearInterval(id);
    }, []);
    return now;
}

function LiveDateTime() {
    const now = useLiveClock();

    const time = now.toLocaleTimeString('en-IN', { hour: '2-digit', minute: '2-digit', second: '2-digit', hour12: true });
    const date = now.toLocaleDateString('en-IN', { weekday: 'short', day: 'numeric', month: 'short' });

    const hour = now.getHours();
    const greeting = hour < 12 ? '🌅' : hour < 17 ? '☀️' : hour < 20 ? '🌆' : '🌙';

    return (
        <div style={{
            display: 'flex', alignItems: 'center', gap: '0.5rem',
            padding: '0.3rem 0.85rem',
            background: 'rgba(255,255,255,0.02)',
            border: '1px solid rgba(255,255,255,0.06)',
            borderRadius: '20px',
            fontFamily: "'Outfit', sans-serif",
            backdropFilter: 'blur(8px)'
        }}>
            <span style={{ fontSize: '0.95rem' }}>{greeting}</span>
            <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'flex-end', lineHeight: 1.2 }}>
                <span style={{ fontSize: '0.82rem', fontWeight: '700', color: '#f1f5f9', letterSpacing: '0.02em', fontVariantNumeric: 'tabular-nums' }}>
                    {time}
                </span>
                <span style={{ fontSize: '0.68rem', color: '#64748b', fontWeight: '500' }}>
                    {date}
                </span>
            </div>
        </div>
    );
}

const Navbar = () => {
    const { theme, toggleTheme } = useTheme();
    const { isAuthenticated, user, logout } = useAuth();
    const [showUserMenu, setShowUserMenu] = useState(false);
    const [scrolled, setScrolled] = useState(false);

    useEffect(() => {
        const handleScroll = () => setScrolled(window.scrollY > 20);
        window.addEventListener('scroll', handleScroll);
        return () => window.removeEventListener('scroll', handleScroll);
    }, []);

    const handleLogout = () => {
        logout();
        setShowUserMenu(false);
    };

    return (
        <nav style={{
            position: 'fixed',
            top: '1rem',
            left: '50%',
            transform: 'translateX(-50%)',
            width: 'calc(100% - 2rem)',
            maxWidth: '1200px',
            padding: '0.5rem 1rem',
            zIndex: 1000,
            transition: 'all 0.4s cubic-bezier(0.23, 1, 0.32, 1)',
            borderRadius: scrolled ? '24px' : '40px',
            backdropFilter: 'blur(20px)',
            background: scrolled ? 'rgba(10, 15, 30, 0.7)' : 'rgba(255, 255, 255, 0.03)',
            border: '1px solid rgba(255,255,255,0.08)',
            boxShadow: scrolled ? '0 20px 40px rgba(0,0,0,0.3)' : 'none'
        }}>
            <div style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
                {/* True FoodBite Premium Animated Aura Logo */}
                <Link to="/" style={{
                    display: 'flex', alignItems: 'center', gap: '0.85rem', 
                    textDecoration: 'none', position: 'relative'
                }}>
                    <style>{`
                        @keyframes premiumSweep {
                            from { transform: rotate(0deg) translateZ(0); }
                            to { transform: rotate(360deg) translateZ(0); }
                        }
                        @keyframes premiumBreathe {
                            0%, 100% { transform: scale(0.95) translateZ(0); opacity: 0.3; }
                            50% { transform: scale(1.1) translateZ(0); opacity: 0.8; }
                        }
                    `}</style>
                    
                    {/* Living Ambient Lens */}
                    <div style={{ position: 'relative', width: '38px', height: '38px', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
                        
                        {/* 1. Deep Core Ambient Glow (The 'Breathe' effect) */}
                        <div style={{
                            position: 'absolute', inset: '-4px', borderRadius: '50%',
                            background: 'linear-gradient(135deg, #16a34a, #0ea5e9)',
                            filter: 'blur(10px)',
                            animation: 'premiumBreathe 6s ease-in-out infinite',
                            willChange: 'transform, opacity',
                            zIndex: 0
                        }} />

                        {/* 2. Slow, Elegant Sweeping Energy Ring */}
                        <div style={{ 
                            position: 'absolute', inset: 0, borderRadius: '50%',
                            background: 'conic-gradient(from 0deg, #0ea5e9, #10b981, #84cc16, #0ea5e9)',
                            animation: 'premiumSweep 12s linear infinite',
                            willChange: 'transform',
                            opacity: 0.9,
                            zIndex: 1
                        }} />

                        {/* 3. Dark Obsidian Glass Core */}
                        <div style={{
                            position: 'absolute', inset: '2px', borderRadius: '50%',
                            background: '#0a0f1e', border: '1px solid rgba(255, 255, 255, 0.08)',
                            display: 'flex', alignItems: 'center', justifyContent: 'center',
                            zIndex: 2,
                            boxShadow: 'inset 0 0 10px rgba(0,0,0,0.8)'
                        }}>
                             <Scan size={18} color="#84cc16" strokeWidth={2.5} />
                             {/* Central AI glowing reticle dot */}
                             <div style={{ position: 'absolute', width: '4px', height: '4px', borderRadius: '50%', background: '#ffffff', boxShadow: '0 0 10px 2px rgba(14, 165, 233, 0.6)' }} />
                        </div>
                    </div>

                    {/* Highly Refined Animated Typography */}
                    <div style={{ display: 'flex', alignItems: 'center', gap: '0.4rem', marginTop: '1px' }}>
                        <style>{`
                            @keyframes textSweep {
                                0% { background-position: 200% 50%; }
                                100% { background-position: -200% 50%; }
                            }
                        `}</style>
                        <span style={{ 
                            fontSize: '1.45rem', fontWeight: '800', color: '#ffffff', 
                            letterSpacing: '-0.02em', lineHeight: 1 
                        }}>
                            True
                        </span>
                        <span style={{ 
                            fontSize: '1.45rem', fontWeight: '900', letterSpacing: '-0.02em', lineHeight: 1,
                            background: 'linear-gradient(90deg, #a3e635 0%, #10b981 25%, #38bdf8 50%, #10b981 75%, #a3e635 100%)',
                            backgroundSize: '200% auto',
                            WebkitBackgroundClip: 'text', WebkitTextFillColor: 'transparent',
                            animation: 'textSweep 6s linear infinite'
                        }}>
                            FoodBite
                        </span>
                    </div>
                </Link>

                {/* Nav Links - Integrated Aesthetic */}
                <div style={{
                    display: 'flex', gap: '2rem', alignItems: 'center',
                    background: 'rgba(255,255,255,0.03)', padding: '0.4rem 2rem',
                    borderRadius: '50px', border: '1px solid rgba(255,255,255,0.05)'
                }}>
                    {['Home', 'Scanner', 'Compare', 'How It Works'].map((label) => (
                        <Link
                            key={label}
                            to={label === 'Home' ? '/' : label === 'Scanner' ? '/scan' : `/${label.toLowerCase().replace(/ /g, '-')}`}
                            style={{
                                color: '#94a3b8', textDecoration: 'none', fontSize: '0.85rem',
                                fontWeight: '700', transition: 'all 0.3s cubic-bezier(0.23, 1, 0.32, 1)',
                                position: 'relative'
                            }}
                            onMouseEnter={e => {
                                e.target.style.color = '#f1f5f9';
                                e.target.style.transform = 'translateY(-1px)';
                            }}
                            onMouseLeave={e => {
                                e.target.style.color = '#94a3b8';
                                e.target.style.transform = 'translateY(0)';
                            }}
                        >
                            {label}
                        </Link>
                    ))}
                </div>

                {/* Right side — theme, auth, datetime */}
                <div style={{ display: 'flex', gap: '0.75rem', alignItems: 'center' }}>
                    <button onClick={toggleTheme} style={{
                        background: 'rgba(255,255,255,0.05)', border: '1px solid rgba(255,255,255,0.08)',
                        padding: '0.5rem', borderRadius: '50%', color: '#f1f5f9',
                        cursor: 'pointer', display: 'flex', alignItems: 'center', transition: 'all 0.3s'
                    }}>
                        {theme === 'dark' ? <Sun size={16} /> : <Moon size={16} />}
                    </button>

                    {isAuthenticated ? (
                        <div style={{ position: 'relative' }}>
                            <button onClick={() => setShowUserMenu(!showUserMenu)} style={{
                                display: 'flex', alignItems: 'center', gap: '0.5rem',
                                padding: '0.4rem 1.2rem',
                                background: 'linear-gradient(135deg, rgba(132, 204, 22, 0.2), rgba(14, 165, 233, 0.2))',
                                border: '1px solid rgba(255,255,255,0.1)', borderRadius: '50px',
                                color: '#f1f5f9', cursor: 'pointer', fontWeight: '800', fontSize: '0.8rem'
                            }}>
                                <User size={14} />
                                <span>{user?.firstName}</span>
                            </button>
                            {/* User menu code... preserved from original */}
                        </div>
                    ) : (
                        <Link to="/login" style={{
                            padding: '0.4rem 1.5rem', color: '#000', textDecoration: 'none',
                            background: '#84cc16', borderRadius: '50px',
                            fontSize: '0.85rem', fontWeight: '800', transition: 'all 0.3s',
                            boxShadow: '0 4px 15px rgba(132, 204, 22, 0.3)'
                        }}
                            onMouseEnter={e => e.target.style.transform = 'scale(1.05)'}
                            onMouseLeave={e => e.target.style.transform = 'scale(1)'}>
                            Login
                        </Link>
                    )}

                    <LiveDateTime />
                </div>
            </div>
        </nav>
    );
};

export default Navbar;
